### PR TITLE
Add the possibility to nest By selectors.

### PR DESCRIPTION
**Pull request by danielthegray**
I have found it useful to build new By selectors by composing/nesting other `By` selectors. I have added a generic method to that for all selector types, as well as a faster way of embedding CSS selectors inside each other (by simply appending the selector strings).

- [x] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

### CODE REVIEW

File: java/client/src/org/openqa/selenium/By.java [hunk 1]
```diff
@@ -138,6 +139,27 @@ public WebElement findElement(SearchContext context) {
...
+   *
+   * @param child The child selector to nest inside the current one (searching for
+   *              elements inside the result of this search).
+   * @return
+   */
+  public By nest(By child) {
+    final By parent = this;
+    return new By() {
+      public List<WebElement> findElements(SearchContext context) {
+        List<WebElement> results = new ArrayList<>();
```
Comment (danielthegray) [inline java/client/src/org/openqa/selenium/By.java:152]: Maybe a `LinkedList` would be better? I'm not sure.

Comment (jleyba) [reply java/client/src/org/openqa/selenium/By.java:152]: No, `LinkedList` is almost never worth using.

**Comment by barancev** [comment]
Please add some tests too

**Review by diemol — COMMENTED** [review]
@danielthegray are you interested in completing this PR? 
Please have a look at the comment above.

**Comment by danielthegray** [comment]
I've had a look and I'm not sure my anonymous By object is the best way to add the feature in a "generic" way. The CSS selector nesting is nice, but the other ones might not work as nicely. I'm open to having this taken over by someone else or coming back to it when I have a bit more time.

**Comment by barancev** [comment]
Let's close this PR for now because it's got stale. We have a class `ByChained` that partially solves the task of combining locators, but if there is a better solution we can consider it, so new PRs are welcome.

**Pull request closed without merging**